File: data/data.code.lua.server/AckRoleRandNameOk.lua
module("AckRoleRandNameOk", package.seeall)


function new()
	local tab_obj = {}
	setmetatable(tab_obj, {__index = AckRoleRandNameOk})
	tab_obj:init()
	return tab_obj
end

function init(self)
	self.uname = ""
end


function encode(self)
	local pack = packet.new()
	pack:writeString(self.uname)
	return pack:encode(Msg.P_ACK_ROLE_RAND_NAME_OK)
end

File: examples/golang/server/proto/ack_role_rand_name_ok.go
package proto

import (
	"packet"
)

type AckRoleRandNameOk struct {
	Uname                    string
}

func (this *AckRoleRandNameOk) Encode() []byte {
	pack := packet.NewWriteBuff(64)

	pack.WriteString(this.Uname)

	return pack.Encode(P_ACK_ROLE_RAND_NAME_OK)
}
